### DISCUSSION TITLE
VSTS build status badge is not showing correctly

### DISCUSSION THREAD

**Discussion by rvhuang** [community/community#22890] [New to GitHub]
<p>The <a href="https://rvhuang.visualstudio.com/MyFirstProject/_apis/build/status/Heuristic.Linq" title="VSTS badge URL" rel="nofollow noopener">VSTS badge URL</a> can be seen on browser directly. However, Markdown cannot correctly display the badge as the response message from <em><a href="http://camo.githubusercontent.com" rel="nofollow noopener">camo.githubusercontent.com</a></em> is always <strong>Invalid upstream response (203)</strong>. See <a href="https://github.com/rvhuang/linq-to-astar" rel="nofollow noopener">my repos</a> for example.</p>
<p>Any advice how to resolve the issue?</p>

**Comment by seveas** [comment, marked as answer]
<p>Looks like one needs to login to see the VSTS badge, so the proxy cannot actually download the image. Only images that do not require authentication can be used in markdown on GitHub.</p>

**Comment by rvhuang** [comment]
<p>Found the answer. The correct URL is formed with VSTS API which does not require log-in. </p>
<p><a href="https://docs.microsoft.com/en-us/rest/api/vsts/build/badge/get%20build%20badge%20data?view=vsts-rest-4.1" rel="nofollow noopener">https://docs.microsoft.com/en-us/rest/api/vsts/build/badge/get%20build%20badge%20data?view=vsts-rest-4.1</a></p>

**Comment by davwheat** [comment]
<p>I only get a setup badge when using the API</p>
